### ISSUE TITLE
Rename add_theme_support( 'disable-custom-font-sizes' ) and default it to true

### ISSUE THREAD

**Issue by mrwweb** [WordPress/gutenberg#12043]
**Is your feature request related to a problem? Please describe.**
Any scenario in which a user needs custom pixel sizes for font-sizing seems fairly niche to me. Is there a specific reason this feature is defaulted to on when there's a good theme-controlled means of enabling a number of variations of font sizing? I propose that the pixel box simply be disabled by default to encourage more use of the primary font-sizing feature.

Ideally, I would pair that with adjusting the theme_support property to be both more descriptive and not falling into the weird situation where `true` = disabled. Maybe: `add_theme_support( 'pixel-font-sizes' )`?

Further, encouraging custom pixel sizes may complicate implementing more responsive sizing (#11671).

**Describe the solution you'd like**

1. Disable custom font sizing in pixels by default.
1. Rename the filter for enabling custom font sizing to make more sense.

**Describe alternatives you've considered**
The existing default, theme-extensible font sizing feels like it will cover most user requirements and encourage more consistency in site designs _and_ likely lead to smoother theme transitions.

Vaguely related: #11671, #9549, #11836

**Comment by mrwweb** [comment]
@karmatosed I get that there's a lot of pressure to not put anything into the 5.0 milestone, but that seems to severely limit future options for this request given that it's a proposal to:

1. Remove UI that's currently on by default
1. Rename a theme_support option

Point 1 feels very much like a "let's turn it off since we can always turn it back on later" situation and Point 2 feels like a necessary change to align the naming convention with all other `add_theme_support()` values _before_ Gutenberg makes it out into the wild and backwards compatibility becomes a stronger prerogative.

Any chance this can get reconsidered or at least some additional eyes for consideration?

**Comment by joyously** [comment]
This should have been done way long time ago (I made comments about it which were ignored also).
+1000

**Comment by melchoyce** [comment]
I almost want core to define a base set of sizes, that can be overridden by themes. If your theme doesn’t add support, you can’t change sizing — and changing font sizing is waaay better than using headings for stylistic purposes.

I'd suggest a core default of:

14
16
18
24
32
48
60
72
96

What do you think, @mrwweb? If this seems like a good direction, I can make a new issue for it.

**Comment by mrwweb** [comment]
@melchoyce I would much prefer just relying on the existing standardized naming scheme that themes can then remap as appropriate.

I believe the default is:

Small - 13px
Normal - 16px
Medium - 20px
Large - 36px
Huge - 48px

I could see adding an "Extra Large" in between Large and Huge, but I'm not clear what the use case is for giving people such fine-grained control over pixel-sizing in a way that probably won't be used consistently from page to page over time.

**Comment by chrisvanpatten** [comment]
See also my pet ticket #11671 :) I think it's really important that we don't lock ourselves into pixels.

**Comment by sarahmonster** [comment]
For accessibility purposes, let's _definitely_ move away from defining these values as pixel sizes. Font sizes set directly in pixels scale with browser zoom, but don't scale for users who set their browser font size to larger than normal (common in older users), which can make for a poor user experience for those who have poor or limited vision.

I'd be a big fan of scaling down the sizes we offer here and making them adjustable by the theme. Maybe something along these lines, based on current sizing (which seems to be using half-pixel sizes as well):

Small: 18px
Medium: 22px (default)
Large: 36px
ExtraLarge: 48px

(...expressing these sizes in rems or ems, of course!)

It's always confused me that we had both "Normal" (ie, the default) and "Medium", so I'm happy to see this list has been scaled down. "Normal" still feels odd in the list, though—"Medium" or "Default" may do a better job of expressing meaning here and helping users choose between the available options. (What's a normal font size anyway?)

I'd also advocate for removing the custom pixel-based font-sizing box. This seems advanced for many users who may not fully understand the units, and who would likely be better serves by a limited palette of font sizes to select from, allow them some freedom of expression in their sites whilst still maintaining consistency and visual harmony. (This is what designers tend to do as well!) Users can always add more fine-grained control via a plugin, via custom CSS classes, other means, but it seems like overkill to offer this in the core product. 

> The rule of thumb is that the core should provide features that 80% or more of end users will actually appreciate and use.
(via [WordPress philosophy](https://wordpress.org/about/philosophy/))

Is there a use case for varying multiple different font sizes in the body copy across a site like this? Do we know how much usage these font size values get? Do 80% of our users require the fine-grained control of deliberate pixel sizes beyond this? 

It seems there are a number of different questions and interacting pieces here (which I may have just inadvertently added to). Here's the approach I'd advise taking here:

1. Convert all pixel sizing to something more easily scaled (rems or ems, depending on the context).
2. Set sensible defaults for named font sizes that are customisable by themes.
3. Remove the custom pixel font-sizing box.

**Comment by joyously** [comment]
Hey Sarah,
I agree, but that's a bit different than this issue. For what you are talking about though, I would propose that there are NO classes for font sizing and the control for font sizing is a range control that generates inline style using a percentage font size. It could have steps so that sizes are easy to reproduce (default to 100%). But if it's in the editor, it should be a one-time thing, not something that the theme or predetermined editor styles are supplying.

**Comment by mapk** [comment]
In [today's Design Team slack triage](https://wordpress.slack.com/archives/C02S78ZAL/p1579023525151500), it was agreed that because this is already "out there in the wild" that we should leave it as is now and close this.

There was a suggestion for a UI that allowed the user to turn this off by @richtabor. Maybe this can be explored in another issue.
